src/components/ProjectDialog.tsx

```diff
@@ -0,0 +1,198 @@
+import { Box, Dialog, Grid, IconButton, Typography, Link } from '@mui/material';
+import { Close } from '@mui/icons-material';
+import TagsList from './TagsList';
+import { Project } from '../types';
+import Carousel from 'react-bootstrap/Carousel';
+import { SetStateAction, useState } from 'react';
+import OasisApprovedIcon from '../assets/OasisApprovedIcon.svg';
+import '../App.css'; 
+import { useTheme } from '@mui/material/styles';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { LanguageMappings } from '../languageUtils';
+
+
+interface ProjectDialogProps {
+  open: boolean;
+  onClose: () => void;
+  project: Project | null;
+  selectedTags: string[];
+  handleTagClick: (tag: string) => void;
+}
+
+const linkStyles = {
+  color: '#3431AC',
+  textDecorationColor: '#3431AC',
+};
+
+const ProjectDialog: React.FC<ProjectDialogProps> = ({
+  open,
+  onClose,
+  project,
+  selectedTags,
+}) => {
+  const [index, setIndex] = useState(0);
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+
+  const handleSelect = (selectedIndex: SetStateAction<number>) => {
+    setIndex(selectedIndex);
+  };
+
+  const getMappedLanguages = (languages: string[]): string[] => {
+    return languages.map((language) => {
+      const mappedLanguage = LanguageMappings[language.toLowerCase() as keyof typeof LanguageMappings];
+      return mappedLanguage ? mappedLanguage : language.substring(0, 1).toUpperCase() + language.substring(1);
+    });
+  };
+  let combinedTags: string[] = [];
+  if (project && project.languages && project.tags) {
+    const mappedLanguages = getMappedLanguages(project.languages);
+    combinedTags = [...project.tags, ...mappedLanguages];
+  }
+  
+  return (
+    <Dialog open={open} onClose={onClose} maxWidth="lg"
+    >
+      <IconButton
+        edge="end"
+        color="inherit"
+        onClick={onClose}
+        style={{ position: 'absolute', top: '32px', right: '32px' }}
+      >
+        <Close />
+      </IconButton>
+   
+      {project && (
+        <div style={{ padding: isMobile ? '32px 20px' : '32px' }}>
+              <Typography variant="h2" sx={{ fontSize: '34px', letterSpacing: '-1.5px', paddingRight: '24px'}}>{project.name}</Typography>
+              <Carousel activeIndex={index} onSelect={handleSelect}
```

Comment:
Could you move the dots showing the screenshots progress *below* the screenshot? Currently it covers it which looks a bit awkward.

Reply:
@matevz - removed.